pagefaultgames/pokerogue
# [Localization] Modification some code about battle stat for localization

## What are the changes?

Modification some code about battle stat for localization.

## Why am I doing these changes?



The message has been shown only english about battle stat such as ATK, DEF. so that it needs to modify that can be localized by translators I think.

## What did change?


- Modification return value of `getBattleStatName`, `getBattleStatLevelChangeDescription` method. it returns localized value used to i18next not English.
- Created `battle-stat.ts` in each locale folders for localizing by translators, they can translate messages about battle stat from this file.
- Created `battle-stat.test.ts` for test localization of battle stat.

### Screenshots/Videos



#### Korean
<img width="828" alt="스크린샷 2024-06-09 오후 1 46 59" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/9989e5c7-ae19-46a1-89db-098232450cc1">

#### Other languages
<img width="908" alt="스크린샷 2024-06-09 오후 1 45 41" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/1f710daa-3b8b-4c80-8606-5df14f530429">

#### Result of test
<img width="665" alt="image" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/9e8ecc31-7268-479d-8517-2dc192659aaf">

## How to test the changes?





- Check out this PR and you can check when you use skill to change battle stat such as growl.
- In NPM, you can test using vitest, `npm run test battle-stat.test.ts`

## Checklist
